src/coreclr/tools/Common/TypeSystem/Common/MethodDesc.cs

```diff
@@ -43,6 +43,8 @@ public struct EmbeddedSignatureData
     /// </summary>
     public sealed partial class MethodSignature : TypeSystemEntity
     {
+        private bool? _isExternalInit;
```

Comment:
This is part of the managed type system shared with crossgen and other tools. We do not want to be adding ILVerify specific state here. The method for checking the IsExternalInit attribute should be an extension method.

Reply:
IL verify is not performance critical. It is fine to run the check every time without caching the result.

Reply:
Yup, I intentionally added it to common thinking it is currently not supported by typesystem and other tools can also use this idiomatic way of checking IsExternalInit rather than working around. It took a while (dumps / binary comparisons) to figure out the semantics. I will move it out of common if it's going to be ilvefiication specific.